starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,155 @@
-// Names and Input
+// 1. Create a variable `hacker1` with the driver's name
 
+var hacker1 = "Samuel"; // driver
 
-//Conditionals
+// 2. Print `"The driver's name is XXXX"` 
+console.log("The driver's name is " + hacker1);
 
+// 3. Create a variable `hacker2` and [ask the user
+var hacker2 = prompt("What's your name ?"); // navigator
 
-// Lorem ipsum generator
+// 4. Print `"The navigator's name is YYYY"`
+console.log("The navigator's name is " + hacker2);
+
+//##
+/*
+5. Depending on which name is longer
+	- `The Driver has the longest name, it has XX characters` or 
+	- `Yo, navigator got the longest name, it has XX characters` or
+	- `wow, you both got equally long names, XX characters!!`
+*/
+
+if(hacker1.length > hacker2.length) {
+	console.log(hacker1 + " has the longest name." );
+} else if(hacker1.length < hacker2.length) {
+	console.log(hacker2 + " has the longest name.");
+} else {
+	console.log("you both got equally long names !");
+}
+
+//## Loops
+/*
+6. Print all the characters of the driver's name, separated by a space and [in capitals]
+ "J O H N"`
+  */
+
+var i = 0;
+var result = "";
+
+while(i < hacker1.length ) {	
+	
+	var str = hacker1.charAt(i);
+	result += " " + str.toUpperCase();
+	i++;
+}
+ 
+console.log(result);
+
+//##
+/*
+7. Print all the characters of the navigator's name, in reverse order "nhoJ"
+  */
+
+var int = 0;
+var resultTwo = [];
+
+while(int < hacker2.length ) {
+	
+	var strTwo = hacker2.charAt(int);
+	resultTwo.push(strTwo);
+
+	int++;
+}
+
+// reverse the array order and join all the items togethers without adding a space to print jessica acissej
+var resultTwoReversed = resultTwo.reverse().join("");
+
+// uppercase last caracter regex https://stackoverflow.com/questions/32149836/how-to-capitalize-the-last-letter-of-each-word-in-a-string
+var hackerNewName = resultTwoReversed.toLowerCase().replace(/[a-z]\b/g, c => c.toUpperCase());
```

Comment:
Nice!! using a regex can be super powerful but also really hard to understand so be careful when using them in your own code